models/src/node/mod.rs

```diff
@@ -29,3 +34,30 @@ pub const K8S_NODE_KIND: &str = "BottlerocketShadow";
 pub const K8S_NODE_PLURAL: &str = "bottlerocketshadows";
 pub const K8S_NODE_STATUS: &str = "bottlerocketshadows/status";
 pub const K8S_NODE_SHORTNAME: &str = "brs";
+
+pub fn read_certificate(public_key_path: &str) -> Result<Vec<u8>> {
+    let mut buf = Vec::new();
+
+    std::fs::File::open(public_key_path)
+        .map_err(|err| Box::new(err) as Box<dyn std::error::Error>)
+        .context(mod_error::IOSnafu)?
+        .read_to_end(&mut buf)
+        .map_err(|err| Box::new(err) as Box<dyn std::error::Error>)
+        .context(mod_error::IOSnafu)?;
+
+    Ok(buf)
+}
+
+pub mod mod_error {
+    use snafu::Snafu;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility(pub))]
+    pub enum Error {
+        #[snafu(display(
+            "IO error occurred while attempting to use APIServerClient: '{}'",
+            source
+        ))]
+        IOError { source: Box<dyn std::error::Error> },
+    }
+}
```

Comment:
Would this error be better placed in the `node/error.rs` file with the rest of this modules errors? (also see comment above where we seem to already have a Result type for this module